### ISSUE TITLE
Fine tuning or Only Policy Training is Possible in TD-MPC2?

### ISSUE THREAD

**Issue by Momorami** [nicklashansen/tdmpc2#64]
Thank you very much for your good work!

I recently read your previous works on fine-tuning and policy training, specifically involving TD-MPC and TD-MPC2:

Fine-tuning Offline World Models in the Real World: https://arxiv.org/abs/2310.16029
PWM: Policy Learning with Multi-Task World Models: https://arxiv.org/abs/2407.02466

I have two questions regarding the TD-MPC2 repository and its potential use for fine-tuning and policy training:

1. Is it possible to fine-tune the pre-trained models that you published for enhance accuracy of single task using this TD-MPC2 repository?
2. Can I train a policy using your pre-trained world model within this repository?

My goal is to either fine-tune or conduct policy training to enhance performance on a specific single task, but I am unsure whether this is achievable using the existing TD-MPC2 repository.

Thank you very much!

**Comment by nicklashansen** [comment]
Hi @Momorami, thanks for your interest in our work!

> Is it possible to fine-tune the pre-trained models that you published

Yes! We have some exploratory finetuning experiments in the paper which indicate that there's something to gain from finetuning a pretrained policy; we use a 19M parameter model trained on 70 tasks for this experiment. The model weights are publicly available but the finetuning code is not. However, it should be fairly straightforward to set up the finetuning, these are roughly the steps:
- Load checkpoint
- Initialize task embedding for your new task to the task embedding of a similar task in the training data
- Run online RL training with default hyperparameters

Let me know if you have any other questions!

**Comment by Momorami** [comment]
Thank you for your kind reply!

Following your outline, I was able to write fine-tuning code and successfully fine-tune the model. 

I tested it on a 5M parameter model, and it reached the same accuracy as training from scratch but in fewer timesteps. 

Thanks again for your great work and fast reply.

**Comment by nicklashansen** [comment]
That's great to hear! Let me know if there's anything else :-)
